updateParamGroups.iLogicVb

```diff
@@ -65,81 +83,80 @@ Public Class updateParamGroup
 	
 		Select Case BeforeOrAfter
 			Case kBefore
-				'MsgBox("kBefore")
 
 			Case kAfter
-				'MsgBox("kAfter")
-				AfterParameterChange(oDoc, parameter)
+				NewName = parameter.Name
+				EditParameterGroup(oDoc)
 
 			Case Else
 				Exit Sub
 
 		End Select
 	End Sub
 
-	Private Sub BeforeParameterChange(oDoc As Inventor.Document, parameter As Inventor.Parameter)
-		'run before the parameter changes, move/delete changed parameters!
+	Private Function HasGroupChanged() As Boolean
+
+		If ParseGroupName(initialName) <> ParseGroupName(NewName) Then
+			Return true
+		Else
+			Return false
+		End If
 
-		initialName = parameter.Name
-		initialGroupName = parseGroupName(initialName)
+	End Function
+
+	Private Sub CleanupParameterGroup(oDoc As Inventor.Document)
+
+		if IsParameterGroupFormat(initialName) = false then
+			exit sub
+		end if
+
+		initialGroupName = ParseGroupName(initialName)
 		GetParamGroups(oDoc)
 
-		If StringContainsGroupName(initialName) = false Then
-			'msgbox("Exiting, StringContainsGroupName returned false")
+		If IsValidGroupName(initialName) = false Then
 			Exit Sub
 		End If
 
-		If DoesGroupNameExist(initialGroupName) = false Then
-			'msgbox("Exiting, DoesGroupNameExist returned false")
-			Exit Sub
-		End If
+		initialGroup = GetGroupByName(initialGroupName)
 
-		If DoesGroupContainParameter(initialName) = false Then
-			'msgbox("Exiting, DoesGroupContainParameter returned false")
+		If DoesGroupContainParameter(NewName, initialGroup) = false Then
```

Comment:
I think this is an extra loop on top of the one for removal, unless the removal's can be avoided somehow